slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/validator/AttestationValidator.java

```diff
@@ -129,6 +134,24 @@ public boolean hasTargetOlderThanWatermark() {
     return false;
   }
 
+  public boolean hasEpochAtOrBeyondHighWatermark() {
+    final Optional<HighWatermark> highWatermark = metadataDao.findHighWatermark(handle);
+    if (highWatermark
+        .map(
+            h ->
+                sourceEpoch.compareTo(h.getEpoch()) >= 0
```

Comment:
this is more of a learning question, why do we need to compare against the source and target epoch?

Reply:
Source and target are both part of the signing request because their interaction can cause slashing conditions, see https://eth2book.info/capella/annotated-spec/#is_slashable_attestation_data

We simplify setting the high watermark (in a similar way that watermark-repair does for low watermark) by setting one epoch value for both source and target, since they can be the same.

In practice, I believe the target should always be >= source, so arguably we could just check the target. I'm not confident enough in the edge cases to make that assumption.

For low watermark, the check is similarly checking both:
```
              || attestationValidator.hasSourceOlderThanWatermark()
              || attestationValidator.hasTargetOlderThanWatermark()
```
For that, perhaps you could also argue you only need to check the source.